src/librustdoc/html/format.rs

```diff
@@ -1212,18 +1210,11 @@ impl clean::Visibility {
                 } else {
                     let path = tcx.def_path(vis_did);
                     debug!("path={:?}", path);
-                    let first_name =
-                        path.data[0].data.get_opt_name().expect("modules are always named");
                     // modified from `resolved_path()` to work with `DefPathData`
                     let last_name = path.data.last().unwrap().data.get_opt_name().unwrap();
                     let anchor = anchor(vis_did, &last_name.as_str(), cache).to_string();
 
-                    let mut s = "pub(".to_owned();
-                    if path.data.len() != 1
-                        || (first_name != kw::SelfLower && first_name != kw::Super)
-                    {
-                        s.push_str("in ");
-                    }
+                    let mut s = "pub(in ".to_owned();
```

Comment:
Why did this change?

Reply:
It seemed impossible for this conditional to evaluate to false. It's checking if the path is exactly equal to `self` or `super`, but there are already conditionals above that check if it's equal to the current module or the parent module.

Removing it also didn't break any test cases, supporting my dead-code theory.

Reply:
That makes sense, thanks. I think I didn't realize what it was doing when @camelid added the `pub(self)` handling back in https://github.com/rust-lang/rust/pull/80368.